src/hotspot/share/oops/cpCache.hpp

```diff
@@ -38,196 +38,90 @@
-//
-// The flags after TosState have the following interpretation:
-// bit 27: 0 for fields, 1 for methods
-// f  flag true if field is marked final
-// v  flag true if field is volatile (only for fields)
-// f2 flag true if f2 contains an oop (e.g., virtual final method)
-// fv flag true if invokeinterface used for method in class Object
-//
-// The flags 31, 30, 29, 28 together build a 4 bit number 0 to 16 with the
-// following mapping to the TosState states:
-//
-// btos: 0
-// ztos: 1
-// ctos: 2
-// stos: 3
-// itos: 4
-// ltos: 5
-// ftos: 6
-// dtos: 7
-// atos: 8
-// vtos: 9
-//
-// Entry specific: field entries:
-// _indices = get (b1 section) and put (b2 section) bytecodes, original constant pool index
-// _f1      = field holder (as a java.lang.Class, not a Klass*)
-// _f2      = field offset in bytes
-// _flags   = field type information, original FieldInfo index in field holder
-//            (field_index section)
-//
-// Entry specific: method entries:
-// _indices = invoke code for f1 (b1 section), invoke code for f2 (b2 section),
-//            original constant pool index
-// _f1      = Method* for non-virtual calls, unused by virtual calls.
-//            for interface calls, which are essentially virtual but need a klass,
-//            contains Klass* for the corresponding interface.
-//            for invokedynamic and invokehandle, f1 contains the adapter method which
-//            manages the actual call. The appendix is stored in the ConstantPool
-//            resolved_references array.
-//            (upcoming metadata changes will move the appendix to a separate array)
-// _f2      = vtable/itable index (or final Method*) for virtual calls only,
-//            unused by non-virtual.  The is_vfinal flag indicates this is a
-//            method pointer for a final method, not an index.
-// _flags   = has local signature (MHs and indy),
-//            virtual final bit (vfinal),
-//            parameter size (psize section)
-//
-// Note: invokevirtual & invokespecial bytecodes can share the same constant
-//       pool entry and thus the same constant pool cache entry. All invoke
-//       bytecodes but invokevirtual use only _f1 and the corresponding b1
-//       bytecode, while invokevirtual uses only _f2 and the corresponding
-//       b2 bytecode.  The value of _flags is shared for both types of entries.
-//
-// The fields are volatile so that they are stored in the order written in the
-// source code.  The _indices field with the bytecode must be written last.
-
 class CallInfo;
 class ResolvedFieldEntry;
 class ResolvedIndyEntry;
+class ResolvedMethodEntry;
 
-class ConstantPoolCacheEntry {
-  friend class VMStructs;
-  friend class ConstantPool;
-  friend class InterpreterRuntime;
+// A constant pool cache is a runtime data structure set aside to a constant pool. The cache
+// holds interpreter runtime information for all field access and invoke bytecodes. The cache
```

Comment:
I think there is a bit more nuance needed here. Strictly, the cpCache *is* a cache since it retains values for re-use that would otherwise need to be recomputed. What it is not is a traditional key-value lookup store i.e. it does not conform to the common usage of 'cache' in computer science parlance but to a more broad use of the term in English. I agree that we could do with a better name.

Also, while you are correct to note that the runtime uses the cpCache during link resolution and that the JITs use the cpCache when generating compiled code, it is also true that the resulting JITted method code does not refer to the cache as part of its operation. So, as far as the execution engine of the JVM is concerned, the info in the cpCache is present to optimize interpreted execution and is incidental to compiled execution. That distinction explains why it has been referred to as an interpreter 'cache' or 'scratch pad' and I think this is worth noting, albeit at the risk of down-playing those other uses. This present discussion may well all that is needed to provide that note so, as you say, the comment should probably avoid tying the cpCache exclusively to the interpreter.